### PR TITLE
[baremetal] Sideport the variable IO buffer size feature to baremetal

### PR DESCRIPTION
**Pull request by AndrzejKurek**
This PR sideports the variable buffer sizes feature along with a part of the tests to baremetal.

### CODE REVIEW

**Comment by AndrzejKurek** [comment]
1056 bytes of code size added.

File: library/ssl_tls.c [hunk 1]
```diff
@@ -8761,6 +8812,66 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
         ssl->handshake = mbedtls_calloc( 1, sizeof(mbedtls_ssl_handshake_params) );
     }
 
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    /* If the buffers are too small - reallocate */
```
Comment (shelib01) [inline library/ssl_tls.c:8816]: @AndrzejKurek , I try to understand the flow of too small buffers. Could you explain when we will want to increase the buffers again ?

Comment (AndrzejKurek) [reply library/ssl_tls.c:8816]: It's before each handshake that we want to have full sized buffers. After the handshake they can be shrinked down again. Renegotiation also forces buffer resizing, as the MFL (maximum fragment length) might change, and this affects buffer sizes.

**Review by shelib01 — APPROVED** [review]

**Pull request merged**